pkg/issuer/vault/issue.go

```diff
@@ -205,33 +205,38 @@ func (v *Vault) requestVaultCert(commonName string, altNames []string, csr []byt
 
 	err = request.SetJSONBody(parameters)
 	if err != nil {
-		return nil, fmt.Errorf("error encoding Vault parameters: %s", err.Error())
+		return nil, nil, fmt.Errorf("error encoding Vault parameters: %s", err.Error())
 	}
 
 	resp, err := client.RawRequest(request)
 	if err != nil {
-		return nil, fmt.Errorf("error signing certificate in Vault: %s", err.Error())
+		return nil, nil, fmt.Errorf("error signing certificate in Vault: %s", err.Error())
 	}
 
 	defer resp.Body.Close()
 
 	vaultResult := certutil.Secret{}
 	resp.DecodeJSON(&vaultResult)
 	if err != nil {
-		return nil, fmt.Errorf("unable to decode JSON payload: %s", err.Error())
+		return nil, nil, fmt.Errorf("unable to decode JSON payload: %s", err.Error())
 	}
 
 	parsedBundle, err := certutil.ParsePKIMap(vaultResult.Data)
 	if err != nil {
-		return nil, fmt.Errorf("unable to parse certificate: %s", err.Error())
+		return nil, nil, fmt.Errorf("unable to parse certificate: %s", err.Error())
 	}
 
 	bundle, err := parsedBundle.ToCertBundle()
 	if err != nil {
-		return nil, fmt.Errorf("unable to convert certificate bundle to PEM bundle: %s", err.Error())
+		return nil, nil, fmt.Errorf("unable to convert certificate bundle to PEM bundle: %s", err.Error())
+	}
+
+	var caPem []byte = nil
+	if len(bundle.CAChain) > 0 {
+		caPem = []byte(bundle.CAChain[0])
 	}
 
-	return []byte(bundle.ToPEMBundle()), nil
+	return []byte(bundle.ToPEMBundle()), caPem, nil
```

Comment:
As far as I can tell its that call to `ToPemBundle()` that packages the private key up with the issued certificate and the entire certificate chain. The thing is, I didnt touch that and Id be very surprised if it made it though testing without being caught

Reply:
Which endpoint are you using during your tests?

See: https://github.com/jetstack/cert-manager/issues/790

I believe you may be using the incorrect endpoint for Certificates!

Reply:
@munnerz  Pretty sure you're right, this is my issuer config:

```
apiVersion: certmanager.k8s.io/v1alpha1
kind: Issuer
metadata:
  name: vault-issuer
  namespace: cert-manager
spec:
  vault:
    auth:
      tokenSecretRef:
        name: cert-manager-vault-token
        key: token
    path: pki/issue/example-dot-com
    server: http://vault:8200
```

so we'll leave it held until I get home and can test with the `sign` endpoint

Reply:
Sorry I spaced out on this, it was just the endpoint and things are working when I switch to using `sign`, this is good to go